notification/app/notification/services/guardian/GuardianNotificationSender.scala

```diff
@@ -60,6 +60,18 @@ class GuardianNotificationSender(
     }
   }
 
+  private def countRegistration(platform: Platform, topics: List[Topic]): Future[Int] = {
+    // in case of an exception when calling registrationCounter, we want to continue anyway
+    registrationCounter.count(topics).map(t => t.counts(platform)).recover {
+      case NonFatal(e) if topics.exists(_.`type` == TopicTypes.Breaking) && platform != Newsstand =>
+        logger.error("Unable to count registration for a list of topics during a breaking news", e)
+        1500000 // fallback on 1.5M: the worse case
+      case NonFatal(e) =>
+        logger.error("Unable to count registration for a list of topics", e)
+        1
+    }
+  }
+
```

Comment:
+1 for the logic but it is related to the number of shard we want to create and therefore rather than having it decoupled from the shard count I would return a `Future[Option[Int]]` here and have the logic in `prepareBatch`:
```
def prepareBatch(platform: Platform, notification: Notification, registrationCount: Option[Int]) = {
     ...
     val count = registrationCount match {
           case None if notification.topics.exists(_.`type` == TopicTypes.Breaking) && platform != Newsstand =>
                logger.error("Unable to count registration for a list of topics during a breaking news", e)
                1500000 // if breaking news we assume everybody is subscrided to it
           case None =>
                logger.error("Unable to count registration for a list of topics", e)
                1
      case Some(c) => c
   }
   shard(count).map { shard =>
   ...
```